src/test/frontend/deploy/createsecret_controller_test.js

```diff
@@ -0,0 +1,110 @@
+  let httpBackend;
+
+  beforeEach(() => {
+    angular.mock.module(deployModule.name);
+    angular.mock.inject(($controller, $httpBackend) => {
+      ctrl = $controller(CreateSecretController, {'namespace': 'default'});
+      httpBackend = $httpBackend;
+    });
+  });
+
+  /**
+   * Test the secret name validation pattern.
+   */
+  it('should validate proper secret names as correct', () => {
+    // given the k8s rule for secret names
+    let rule = ctrl.secretNamePattern;
+
+    // then the following secret names should be accepted
+    expect("mysecret".match(rule)).toBeDefined();
+    expect("my.secret.com".match(rule)).toBeDefined();
+    expect("my-secret32.com".match(rule)).toBeDefined();
+  });
+
+  it('should validate inproper secret names as incorrect', () => {
+    // given the k8s rule for secret names
+    let rule = ctrl.secretNamePattern;
+
+    // then the following secret names should be rejected
+    expect("illegalsecret§$".match(rule)).toBeNull();
+    expect(".dotpreffix.com".match(rule)).toBeNull();
+    expect("dotsuffix.com.".match(rule)).toBeNull();
+    expect("-dashprefix.com".match(rule)).toBeNull();
+    expect("dashsuffix-".match(rule)).toBeNull();
+    expect("umlautsÖÄ.not.allowed".match(rule)).toBeNull();
+    expect("name with spaces".match(rule)).toBeNull();
+    expect("  ".match(rule)).toBeNull();
+  });
+
+  /**
+   * Test the Base64 validation pattern for the .dockercfg data.
+   */
+  it('should validate Base64 data as correct', () => {
+    // given a Base64 matching regex
+    let rule = ctrl.dataPattern;
+    // then the following data should be accepted
+    expect(
+        `eyAiaHR0cHM6Ly9pbmRleC5kb2NrZXIuaW8vdjEvIjogeyAiYXV0aCI6ICJabUZyWlhCaGMzTjNiM0prTVRJSyIsICJlbWFpbCI6ICJqZG9lQGV4YW1wbGUuY29tIiB9IH0K`
+            .match(rule))
+        .toBeDefined();
+  });
+
+  it('should validate non-Base64 data as incorrect', () => {
+    // given a Base64 matching regex
+    let rule = ctrl.dataPattern;
+
+    // then the following data strings should be rejected
+    expect("aaa".match(rule)).toBeNull();
+    expect("=asd8".match(rule)).toBeNull();
+    expect("==aaa".match(rule)).toBeNull();
+    expect("abcde".match(rule)).toBeNull();
+  });
+
+  it('should not submit if the form has validation errors', () => {
+    // given a validation error in the html form
+    ctrl.secretForm = {};
+    ctrl.secretForm.$valid = false;
+    // when trying to submit
+    ctrl.createSecret();
+    // then form data was not sent to backend (thus flush will throw error)
+    expect(httpBackend.flush).toThrow();
+  });
+
+  it('should submit if the form has no valiadtion errors', () => {
+    // given a validation error in the html form
+    ctrl.secretForm = {};
+    ctrl.secretForm.$valid = true;
+    ctrl.secretName = "mysecret";
+    ctrl.data =
+        `eyAiaHR0cHM6Ly9pbmRleC5kb2NrZXIuaW8vdjEvIjogeyAiYXV0aCI6ICJabUZyWlhCaGMzTjNiM0prTVRJSyIsICJlbWFpbCI6ICJqZG9lQGV4YW1wbGUuY29tIiB9IH0K`;
```

Comment:
Our lint rules forbid this, should I put the parts in variables and use the ${..} syntax?